src/Kiota.Builder/Writers/Python/CodeMethodWriter.cs

```diff
@@ -809,11 +819,11 @@ private string GetSerializationMethodName(CodeTypeBase propType)
             _ => "write_object_value",
         };
     }
-    internal string GetTypeFactory(bool isVoid, bool isStream, bool isEnum, string returnType)
+    internal string GetTypeFactory(bool isVoid, bool isStream, bool isEnum, string returnType, bool isCollection)
     {
         if (isVoid) return string.Empty;
         if (isStream || isEnum) return $" \"{returnType}\",";
-        if (conventions.IsPrimitiveType(returnType)) return $" {returnType},";
+        if (conventions.IsPrimitiveType(returnType)) return isCollection ? $" {returnType}," : $" \"{returnType}\",";
```

Comment:
can you expand on why we have a difference between singular and collection properties? shouldn't it effectively be the same? (i.e. the needed type information is identical, the difference is that we're looping for collections)

Reply:
Thanks for raising this. I agree with this. The type info should be identical. 
However, this alignment looks to be something we missed in building of the abstraction libraries and changing this now would be breaking change (Will create an issue for fixing this next major version once we merge this).

implementation of send_primitive - https://github.com/microsoft/kiota-python/blob/db64f963bd94a6d106a847ea3112d5799bb72568/packages/http/httpx/kiota_http/httpx_request_adapter.py#L300
implementation of send_collection_of_primitive_async - https://github.com/microsoft/kiota-python/blob/db64f963bd94a6d106a847ea3112d5799bb72568/packages/http/httpx/kiota_http/httpx_request_adapter.py#L250 

This is why after the change made in the PR at https://github.com/microsoft/kiota/pull/5616, it ended up breaking resulting in the issue at https://github.com/microsoft/kiota/issues/5744

Reply:
so we effectively shipped a breaking change last month, the original situation was `"str"` for both, and now collection takes `str`. Correct?
If so, we should look at reverting the breaking change, I'm surprised nobody has called us out on this one yet.

Reply:
Sorry, I wasn't clear here.

The implementation was always inconsistent for both implementations. 
If you take a look at the implementation for `send_primitive`, you'll see that it expects [string literals](https://github.com/microsoft/kiota-python/blob/db64f963bd94a6d106a847ea3112d5799bb72568/packages/http/httpx/kiota_http/httpx_request_adapter.py#L338). 

However, the implementation for `send_collection_of_primitive_async` expects [types](https://github.com/microsoft/kiota-python/blob/db64f963bd94a6d106a847ea3112d5799bb72568/packages/http/httpx/kiota_http/httpx_request_adapter.py#L287), so making them consistent would be the breaking change. 
So, we need to align on this for now and align on the next breaking change as switching from one side to the other fixes one and breaks the other as seen in https://github.com/microsoft/kiota/pull/5616

Reply:
Thanks for clarifying, can you please create an issue to capture that on the python repo?

Reply:
Created https://github.com/microsoft/kiota-python/issues/408 to fix this at the next major version